### PR TITLE
use project coordinates for dependency in editor

### PR DESCRIPTION
**Pull request by MauricioUyaguari**
se `depot` server to query for available dependency projects. Switch to adding dependency using project coordinates consisting of `groupId` and `artifactId`.

| Q                       | A                                |
| ----------------------- | -------------------------------- |
| Fixed issues?           |  |
| Tests added + passed?   | Yes                              |
| Changeset added?        | Yes                              |
| Documentation PR link   |                                  |
| Any dependency changes? | No                               |

### CODE REVIEW

**Comment by maoo** [comment]
@cla-bot[bot] check

File: .changeset/grumpy-birds-agree.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-studio": patch
+---
+
+Use `depot` server to query for available dependency projects.  Switch to adding dependency using project coordinates consisting of `groupId` and `artifactId`.
```
Comment (akphi) [inline .changeset/grumpy-birds-agree.md:5]:
```suggestion
Use `depot` server to query for available dependency projects.  Switch to adding dependencies using project coordinates consisting of `groupId` and `artifactId`. 
```

**Comment by akphi** [comment]
@MauricioUyaguari should we start bringing in the concept of `MetamodelDependency` to match with `SDLC`?

**Comment by MauricioUyaguari** [comment]
> @MauricioUyaguari should we bring in the concept of `MetamodelDependency` to match with `SDLC`?

No, I believe this is deprecated in SDLC and definitely not used in studio anywhere. The model used is `ProjectDependency`.

**Review by akphi — APPROVED** [review]
Approved, but as you have mentioned, we need to coordinate this with the release and deployment of SDLC, so let's not merge it yet

**Pull request merged**